### DISCUSSION TITLE
Discord bot on python

### DISCUSSION THREAD

**Discussion by ghost** [TheAlgorithms/Python#8830] [General]
How to create good discord.bot on python like i have server and i don't know how to create a good discord bot for my friends server

**Comment by ylmzfurkan** [comment]
Hii @PoProstuMadryPl 

To create a Discord bot using Python and discord.py library, follow these steps:

Install Python on your computer and create a project directory.

Set up a virtual environment in the project directory to isolate dependencies. Activate the virtual environment.

Install the discord.py library by running the command: pip install discord.py.

Go to the Discord Developer Portal (https://discord.com/developers/applications) and create a new application.

In the application settings, navigate to the "Bot" tab and create a new bot. Set the bot name and profile picture.

Copy the bot token from the "Token" section. This token is a secret key for authenticating your bot. Keep it secure and don't share it.

Create a Python file in your project directory and write the following code to create your Discord bot:

```
import discord
from discord.ext import commands

bot = commands.Bot(command_prefix='!')

@bot.event
async def on_ready():
    print(f'We have logged in as {bot.user}')

@bot.command()
async def hello(ctx):
    await ctx.send('Hello, World!')

bot.run('YOUR_BOT_TOKEN')
```

Replace 'YOUR_BOT_TOKEN' with the bot token you copied earlier.

Save the code and run the Python file.

Your bot will log in and join the Discord server. You can use the prefix '!' we set to issue commands. Try using the '!hello' command in the Discord server to receive a "Hello, World!" message from your bot.

**Reply by ghost** [reply]
@ylmzfurkan 
Thank you very much, now I can make a decent bot for my friends' server

**Reply by ylmzfurkan** [reply]
You're welcome. I will be glad if you share the result with me.
